src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -444,6 +491,53 @@ private static boolean hasLineBreakBefore(DetailAST rightCurly) {
     }
 
     /**
+     * Checks if right curly is alone on line in token the lambda.
+     * @param detail for validation.
+     * @param <T> token type.
+     * @return true, if right curly is alone on line.
+     */
+    private static <T> boolean isAloneOther(DetailsType<T> detail) {
+        final Details details = (Details) detail.getTokenDetails();
+        final DetailAST lcurly = details.lcurly;
+        final DetailAST rcurly = details.rcurly;
+        final DetailAST previousSibling = rcurly.getPreviousSibling();
+        final int lcurlyNo = lcurly.getLineNo();
+        final int rcurlyNo = rcurly.getLineNo();
+        boolean result = false;
+        boolean result2 = false;
+        if (lcurlyNo != rcurlyNo) {
+            if (previousSibling == null) {
+                result = true;
+            }
+            else if (rcurlyNo != previousSibling.getLineNo()) {
+                result2 = true;
+            }
+        }
+        return result || result2;
+    }
+
+    /**
+     * Structure that contains token type.
+     * @param <T> token type
+     */
+    private static final class DetailsType<T> {
```

Comment:
@MEZk Even though class is private, we could still extend it in this same class (`RightCurlyCheck`) if there was no final.
I'm not sure which TC violation required the `final`.